integrationTest: Add IT for user_token grant variants

strehle opened the pull request:
User Token Grant Type is proprietary grant type. Until now we dont have pure integration tests. therefore add with this PR one.
https://docs.cloudfoundry.org/api/uaa/version/76.18.0/index.html#user-token-grant

In PR 
https://github.com/cloudfoundry/uaa/pull/2193
I noticed that no integration test has to be change but the PR is a restriction of current behavior

Because of the other PR I named the tests according to client types, e.g. public and/or confidential. A public client is a client without any secret or with an empty secret (like cf client). A confidential client is a client with a secure secret.

strehle committed:
Add IT for user_token grant

Tallicia commented:
Can we get a description on what this is addressing?

strehle commented:
> Can we get a description on what this is addressing?

created it to better understand what https://github.com/cloudfoundry/uaa/pull/2193 changes, so if we proceed with it, this test should be merged

swalchemist commented on uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/feature/UserTokenGrantIT.java:
```diff
@@ -0,0 +1,111 @@
+    }
+    webDriver.get(appUrl + "/j_spring_security_logout");
+    webDriver.manage().deleteAllCookies();
+  }
```
Missing a line break here.

strehle replied:
added

swalchemist commented on uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/feature/UserTokenGrantIT.java:
```diff
@@ -0,0 +1,111 @@
+  final String user_token_secret = "secret";
+
+  @Before
+  @After
```
Really like seeing the @Before and @After together here for cleanup.

strehle replied:
after is enough

swalchemist replied:
I think that's the first time I saw code being changed because I praised it. :)

swalchemist commented on uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/feature/UserTokenGrantIT.java:
```diff
@@ -0,0 +1,111 @@
+    webDriver.manage().deleteAllCookies();
+  }
+  @Test
+  public void testPasswordTokenDoUserTokenGrant() {
```
Confusing method name. Can't tell what to expect to find in this test from the name.

strehle committed:
Merge branch 'develop' of github.com:cloudfoundry/uaa into userTokenIntegrationTest

strehle committed:
add more examples for user_token grant

The grant is similar to refresh but always without a secret.
The 4 examples are for the discussion about issue https://github.com/cloudfoundry/uaa/pull/2193

strehle committed:
refactor test cases

strehle committed:
Merge branch 'develop' of github.com:cloudfoundry/uaa into userTokenIntegrationTest

strehle committed:
test renamed

swalchemist commented:
@strehle Does #2193 need to be merged before this one?

strehle commented:
> @strehle Does #2193 need to be merged before this one?

no the other way around, this should be merged first and then #2193 (after rebase) because then in PR you see what is not working anymore

swalchemist commented on uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/feature/UserTokenGrantIT.java:
```diff
@@ -0,0 +1,190 @@
+    String newToken = doUserTokenGrant(user_token_id, token, HttpStatus.OK);
+
+    // Then validation expected result
+    assertNotNull(newToken);
```
Why not move all the assertNotNull() calls to checkRefreshToken()?

swalchemist commented on uaa/src/test/java/org/cloudfoundry/identity/uaa/integration/feature/UserTokenGrantIT.java:
```diff
@@ -0,0 +1,190 @@
+    String newToken = doUserTokenGrant("admin", token, HttpStatus.UNAUTHORIZED);
+
+    // Then validation expected result
+    assertNull(newToken);
```
This assertNull() is only testing functionality in the test function doUserTokenGrant(). Not really needed here.

swalchemist reviewed (approved):
Just a few more minor comments. I'll go ahead and approve so you can take a look and then move ahead.

strehle committed:
review
